src/rdata/caa.rs

```diff
@@ -0,0 +1,649 @@
+//! Record data from [RFC 8659]: CAA records.
+//!
+//! This RFC defines the CAA record type.
+//!
+//! [RFC 8659]: https://www.rfc-editor.org/info/rfc8659
+
+use crate::base::{
+    name::FlattenInto,
+    rdata::ComposeRecordData,
+    scan::{Scan, Scanner, ScannerError},
+    wire::{Compose, Parse, ParseError},
+    zonefile_fmt::{self, Formatter, ZonefileFmt},
+    CanonicalOrd, CharStr, ParseRecordData, RecordData, Rtype,
+};
+use core::{cmp::Ordering, fmt, hash};
+#[cfg(feature = "serde")]
+use octseq::{
+    builder::{EmptyBuilder, FromBuilder},
+    serde::DeserializeOctets,
+    serde::SerializeOctets,
+};
+use octseq::{Octets, OctetsBuilder, OctetsFrom, OctetsInto, Parser};
+
+//------------ Caa ---------------------------------------------------------
+
+/// Caa record data.
+///
+/// The Certification Authority Authorization (CAA) DNS Resource Record allows
+/// a DNS domain name holder to specify one or more Certification Authorities
+/// (CAs) authorized to issue certificates for that domain name.
+///
+/// CAA Resource Records allow a public CA to implement additional controls to reduce the
+/// risk of unintended certificate mis-issue.
+///
+/// The Caa record type is defined in [RFC 8659, section 4.1][1].
+///
+/// [1]: https://www.rfc-editor.org/rfc/rfc8659#section-4.1
+#[derive(Clone)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(bound(
+        serialize = "
+            Octs: SerializeOctets + AsRef<[u8]>
+        ",
+        deserialize = "
+            Octs: FromBuilder + DeserializeOctets<'de>,
+            <Octs as FromBuilder>::Builder:
+                OctetsBuilder + EmptyBuilder
+                + AsRef<[u8]>,
+        ",
+    ))
+)]
+pub struct Caa<Octs> {
+    flags: u8,
+    tag: CaaTag<Octs>,
+    #[cfg_attr(
+        feature = "serde",
+        serde(
+            serialize_with = "octseq::serde::SerializeOctets::serialize_octets",
+            deserialize_with = "octseq::serde::DeserializeOctets::deserialize_octets",
+            bound(
+                serialize = "Octs: octseq::serde::SerializeOctets",
+                deserialize = "Octs: octseq::serde::DeserializeOctets<'de>",
+            )
+        )
+    )]
+    value: Octs,
+}
+
+impl Caa<()> {
+    // The rtype of this record data type.
+    pub const RTYPE: Rtype = Rtype::CAA;
+}
+
+impl<Octs> Caa<Octs> {
+    /// Creates a new CAA record data from the flags, tag, and value.
+    pub fn new(flags: u8, tag: CaaTag<Octs>, value: Octs) -> Self {
+        Caa { flags, tag, value }
+    }
+
+    /// If the value is set to "1", the Property is critical.
+    /// A CA MUST NOT issue certificates for any FQDN if the
+    /// Relevant RRset for that FQDN contains a CAA critical
+    /// Property for an unknown or unsupported Property Tag.
+    pub fn flags(&self) -> u8 {
+        self.flags
+    }
+
+    /// The Property identifier
```

Comment:
This should be an active phrase, e.g., “Returns the property identifier.”

The same goes for the other methods. They should all have a docstring at least with the leading line, even the private ones.